### ISSUE TITLE
JUnit should process annoations in package-info.java

### ISSUE THREAD

**Issue by brianegge** [junit-team/junit4#670]
It would be useful to be able to add package level annotations to package-info.java. For example adding @org.junit.Ignore to the package should ignore all tests in that package.

**Comment by adrianosimoes** [comment]
Is this still a valid issue?

**Comment by kcooney** [comment]
@adrianosimoes I don't think the maintainers are thrilled with the idea of processing package-level annotations. I think we should leave it open in case someone has a use case other than `@Ignore` where it would be useful to process package-level annotations. If `@Ignore` is the only supported feature, then it doesn't seem very compelling IMHO.

**Comment by kcooney** [comment]
One use case just came up (`@RunWith`). I could imagine wanting to specify `@Category` at a package level as well. For `@RunWith` it is clear what happens when the package has the annotation and some classes in the package are also annotated. For `@Category` the desired behavior is less clear.

**Comment by brianegge** [comment]
Supporting package level annotations would be straight forward to do. However, the number of people actually using them would likely be between zero and one. It appears there is little demand for this feature, so it's probably best to close this request.

**Comment by kcooney** [comment]
There is discussion about possibly using them in #1423

**Comment by cesar1000** [comment]
@brianegge I disagree. A package-level @Ignore annotation as you suggested in your original request can be really useful, as there's no convenient way to disable a set of test un current JUnit. So is a @RunWith annotation for defining a default runner for a set of tests.

As to whether this would see adoption: maybe not if annotations need to be added manually, but IDEs could use this feature and expose it through a high level interface.

**Comment by kcooney** [comment]
From a discussion in #1423 

> it appears that Java will only initialize `Package`
> objects when first loading a class from the package.
> This is ok if annotating the package containing the
> tests, but won't work with parent packages (unless
> adding placeholder classes to these packages and
> loading them explicitly, but that's a bad hack).
